src/pwd/pwd.c

```diff
@@ -1,6 +1,95 @@
+#include <limits.h>
+#include <stdbool.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 
-int main(void) {
-  printf("Hello, World!\n");
-  return 0;
-}
\ No newline at end of file
+#define NAME "pwd (canoutils)"
+#define VERSION "1.0.0"
+#define AUTHOR "Mehadav Itay (whiteapolo)"
+
+#define print_version()                                                        \
+  do {                                                                         \
+    printf("%s\nversion: %s\nby: %s\n", NAME, VERSION, AUTHOR);                \
+  } while (0)
+
+char *realpath(const char *path, char *resolved_path);
+
+int status = 0;
+bool logical = false;
+
+char buff[256];
+
+void print_pwd_pysical() {
+  getcwd(buff, 256);
+  printf("%s\n", buff);
+}
+
+void print_pwd_logical() {
+  char *env = getenv("PWD");
+  if (env == NULL) {
+    printf("pwd: cannot access PWD varibale\n");
+    status = 1;
+    return;
+  }
+  printf("%s\n", env);
+}
+
+void print_file_path(const char *file) {
+  if (access(file, F_OK)) {
+    printf("pwd: cannot access '%s': No such file or directory\n", file);
+    return;
+    status = 1;
+  }
+
+  realpath(file, buff);
+  printf("%s\n", buff);
+}
+
+void print_help() {
+  puts(" -L, get pwd from environment varible");
+  puts(" --version, output version information and exit");
+  puts(" --help, display this help and exit");
+}
+
+int main(int argc, char **argv) {
+  int option_count = 0;
+  for (int i = 1; i < argc; i++) {
+    if (*argv[i] == '-') {
+      if (argv[i][1] == '-') {
+        if (!strcmp(argv[i], "--version")) {
```

Comment:
Since `strcmp` call already check the presence of `--` you don't need to add those extra conditions.